panther_diagnostics/test/unit/test_filesystem.cpp

```diff
@@ -0,0 +1,81 @@
+#include <gtest/gtest.h>
+#include <fstream>
+
+#include "panther_diagnostics/filesystem.hpp"
+
+class TestFilesystem : public testing::Test
+{
+public:
+  TestFilesystem();
+
+  std::string CreateTestFile(const std::string & content);
+  void RemoveTestFile(const std::string & file_path);
+
+protected:
+  std::shared_ptr<panther_diagnostics::Filesystem> filesystem_;
+
+  static constexpr char kDummyString[] = "Hello World!";
+};
+
+TestFilesystem::TestFilesystem() : filesystem_(std::make_shared<panther_diagnostics::Filesystem>())
+{
+}
+
+std::string TestFilesystem::CreateTestFile(const std::string & content)
+{
+  std::string file_path = testing::TempDir() + "test_file.txt";
+
+  std::ofstream file(file_path);
+  file << content;
+  file.close();
+
+  return file_path;
+}
+
+void TestFilesystem::RemoveTestFile(const std::string & file_path)
+{
+  std::remove(file_path.c_str());
+}
+
+TEST_F(TestFilesystem, ReadFileSuccess)
+{
+  const auto test_file_path = CreateTestFile(kDummyString);
+
+  auto content = filesystem_->ReadFile(test_file_path);
+
+  EXPECT_STREQ(kDummyString, content.c_str());
+
+  RemoveTestFile(test_file_path);
+}
+
+TEST_F(TestFilesystem, ReadFileNonExistent)
+{
+  // File does not exist
+  auto const test_file_path = testing::TempDir() + "test_file.txt";
```

Comment:
```suggestion
  auto const test_file_path = testing::TempDir() + "test_missing_file.txt";
```
In case file from previous test is not removed correctly, this prevents this test from failing